Code/Tools/SceneAPI/SceneBuilder/SceneImporter.cpp

```diff
@@ -222,7 +222,12 @@ namespace AZ
                     int childCount = node.m_node->GetChildCount();
                     for (int i = 0; i < childCount; ++i)
                     {
-                        std::shared_ptr<AssImpSDKWrapper::AssImpNodeWrapper> child = std::make_shared<AssImpSDKWrapper::AssImpNodeWrapper>(node.m_node->GetChild(i)->GetAssImpNode());
+                        const std::shared_ptr<SDKNode::NodeWrapper> nodeWrapper = node.m_node->GetChild(i);
+                        auto assImpNodeWrapper = std::dynamic_pointer_cast<AssImpSDKWrapper::AssImpNodeWrapper>(nodeWrapper);
```

Comment:
do we really use dynamic_pointer_cast<> in the code normally? Normally we steer away from using the STD dynamic_cast stuff in favor of azrtti_cast<> if possible.